### DISCUSSION TITLE
JoyCaption Alpha One (2024-09-20a)

### DISCUSSION THREAD

**Discussion by SunGreen777** [lllyasviel/stable-diffusion-webui-forge#2080] [Ideas]
UPD > alpha 2 
https://huggingface.co/spaces/fancyfeast/joy-caption-alpha-one
A very good tool, is it possible to add it to Space?
It works like Florence-2 Demo, only better

**Comment by Myoko** [comment]
It seems like a good idea.

**Comment by SillySilk** [comment]
Looks great

**Comment by shodan5000** [comment]
JoyCaption Alpha Two is available as well. Also as a one click installer or through Pinokio.

**Comment by SillySilk** [comment]
tried the one click installer got a error, anyone know where I should be
installing?
C:\caption\joy-caption-alpha-two>app.py
Traceback (most recent call last):
  File "C:\caption\joy-caption-alpha-two\app.py", line 1, in <module>
    import spaces
ModuleNotFoundError: No module named 'spaces'

**Reply by shodan5000** [reply]
Are you using this one click installer?

[https://civitai.com/articles/7801/one-click-installer-for-joycaption-alpha-two-gui-mod](url)

**Reply by shodan5000** [reply]
I just unzipped this one and placed it into an extra storage drive. Click install and let it do its thing.

**Comment by SunGreen777** [comment]
Up

**Comment by ScionoicS** [comment]
TagGUI is better and gives access to multiple models.  Joycaption is suited towards pornographic image captioning more than general captions. You can throw a regular image at it and it'll somehow describe like it's smutty.

**Reply by ostap667inbox** [reply]
> Joycaption is suited towards pornographic image captioning more than general captions. You can throw a regular image at it and it'll somehow describe like it's smutty.

Lol whut? :)
[alpha one](https://huggingface.co/spaces/fancyfeast/joy-caption-pre-alpha), [alpha two](https://huggingface.co/spaces/fancyfeast/joy-caption-alpha-two)
I've been using alpha one and alpha two versions since their release. And not once has an image that does not contain nsfw content been recognized and described so erroneously that the nsfw characteristics in the description were applied.

**Reply by ScionoicS** [reply]
You must've missed the part where the author explicitly describes it to be intended for NSFW captioning.  

https://github.com/fpgaminer/joycaption

**Reply by ostap667inbox** [reply]
> You must've missed the part where the author explicitly describes it to be intended for NSFW captioning.
> 
> https://github.com/fpgaminer/joycaption

1. You must have been inattentive in reading what the author wrote. He nowhere states that this model was created for NSFW captioning. He only says that the model does not have the same censorship restrictions as other major models. You do understand the logical difference between “can be used for NSFW purpose” and “specifically created for NSFW purpose”, right?
2. Also, I repeat, your statement above is completely untrue: non-NSFW images do not receive pornographic descriptions as you claim.
_"You can throw a regular image at it and it'll somehow describe like it's smutty."_ - Did you write this? It's not true.

**Comment by SunGreen777** [comment]
JoyCaption Alpha Two describes the image perfectly, I especially like the critic mode

**Comment by gigend** [comment]
which's better, alpha one or alpha two?
